sci-misc/boinc/files/boinc.init

```diff
@@ -9,46 +9,91 @@ depend() {
 	use dns net ntp-client ntpd
 }
 
-
 create_work_directory() {
-	if [[ ! -d "${RUNTIMEDIR}" ]]; then
+	local sslcrt="/etc/ssl/certs/ca-certificates.crt"
+
+	if [ ! -d "${RUNTIMEDIR}" ]; then
 		einfo "Directory ${RUNTIMEDIR} does not exist, creating now."
 		mkdir -p "${RUNTIMEDIR}"
-		if [[ ! -d "${RUNTIMEDIR}" ]]; then
+		if [ $? -ne 0 ]; then
 			eeror "Directory ${RUNTIMEDIR} could not be created!"
 			return 1
 		fi
 
 		# ensure proper ownership
 		chown "${USER}:${GROUP}" "${RUNTIMEDIR}"
+		if [ $? -ne 0 ]; then
+			eeror "Changing ownership of '${RUNTIMEDIR}' to '${USER}:${GROUP}' failed!"
+			return 1
+		fi
 	fi
 
-	if [[ ! -e "${RUNTIMEDIR}"/ca-bundle.crt ]]; then
-		ln -s /etc/ssl/certs/ca-certificates.crt "${RUNTIMEDIR}"/ca-bundle.crt
+	if [ ! -e "${RUNTIMEDIR}"/ca-bundle.crt ]; then
+
+		if [ -f "${sslcrt}" ]; then
+			ln -s "${sslcrt}" "${RUNTIMEDIR}"/ca-bundle.crt
+			if [ $? -ne 0 ]; then
+				eeror "Symlinking '${sslcrt}' failed!"
+				return 1
+			fi
+		else
+			eerror "'${sslcrt}' does not exist!"
+			return 1
+		fi
 	fi
 
 	return 0
 }
 
+fix_lib_symlinks() {
+	local src="$1"
+	local tgt="$2"
 
-cuda_check() {
-	local libtarget="${RUNTIMEDIR}/libcudart.so"
-	local libsource="$(ls -t /opt/cuda/lib*/libcudart.so 2>/dev/null | head -n 1)"
+	# If the source does not exist, we can not do anything
+	if [ ! -f "${src}" ] ; then
+		return 1
+	fi
 
-	# Remove a broken symlink
-	if [[ -h "${libtarget}" ]] \
-	&& [[ "${libsource}" != "$(readlink "${libtarget}")" ]]; then
-		rm -f "${libtarget}"
+	# Check whether the symlink is already there and in order
+	if [ -L "${tgt}" ] ; then
+		if [ -f "$tgt" ] ; then
+			return 0
+		else
+			# Remove broken symlink
+			rm -f "${tgt}"
+			if [ $? -ne 0 ]; then
+				eeror "Removing '${tgt}' failed!"
+				return 1
+			fi
+		fi
 	fi
 
 	# symlink the correct path
-	if [[ -n "${libsource}" ]] \
-	&& [[ -f "${libsource}" ]] \
-	&& [[ ! -h "${libtarget}" ]]; then
-		ln -snf "$libsource" "${libtarget}"
+	ln -snf "$src" "$tgt"
+	if [ $? -ne 0 ]; then
+		eeror "Symlinking '${src}' to '${tgt}' failed!"
+		return 1
 	fi
 }
 
+cuda_check() {
+	local libsource="$(echo /opt/cuda/lib*/libcudart.so)"
+	local libtarget="${RUNTIMEDIR}/libcudart.so"
+
+	fix_lib_symlinks "$libsource" "$libtarget"
+}
+
+opencl_check() {
+	local libtarget="${RUNTIMEDIR}/libOpenCL.so"
+
+	# which one is it? We can only try to test it.
+	fix_lib_symlinks "/usr/lib/libOpenCL.so"   "$libtarget" && return 0
```

Comment:
This is the 32-bit version on modern amd64 systems (i.e. mine). Hardcoding libdirs is certainly not going to work. Maybe it would be reasonable to put `@libdir@` here, make the script an `.in` and sed the correct libdir in its place inside the ebuild.